File: 0x0E_Sort/selection_sort.py
def sort_func(x):
    n = len(x)
    for i in range(n):
        tmp = i ## 현재 단계에서 맨 앞 자리.
        for j in range(i+1, n):
            if x[tmp] > x[j]:
                tmp = j
        x[i], x[tmp] = x[tmp], x[i]

    return x

if __name__ == "__main__":
    arr = [9, 1, 6, 8, 4, 3, 2, 0]
    print(sort_func(arr))

File: cpp/09-Sort/selection_sort.cpp
#include <iostream>

using namespace std;

int main()
{
    int arr[10] = {3, 2, 7, 116, 62, 235, 1, 23, 55, 77};
    
    for(int i = 0; i < 10-1; i++)
    {
        int min_idx = i;
        for(int j = i+1; j < 10; j++)
        {
            if(arr[min_idx] > arr[j])
                min_idx = j;
        }
        int tmp = arr[i];
        arr[i] = arr[min_idx];
        arr[min_idx] = tmp;

        for(int k = 0; k < 10; k++)
            cout << arr[k] << " ";
        cout << "\n";
    }
}